erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -1356,7 +1356,9 @@ def make_item_gl_entries(self, gl_entries):
 				if item.is_fixed_asset:
 					asset = self.get_asset(item)
 
-					if self.is_return:
+					if (self.docstatus == 2 and not self.is_return) or (
+						self.docstatus == 1 and self.is_return
+					):
```

Comment:
So the new 'if' condition executes only on return document upon submission and not cancellation. Is updating disposal date and reversing depreciation not required for Return cancellation?